### DISCUSSION TITLE
Reevaluating Parent Grinding Faults

### DISCUSSION THREAD

**Discussion by ZenGround0** [filecoin-project/FIPs#402] [Enhancements - Technical]
It's recently been raised that some storage providers are running into situations where they naturally create parent grinding faults when mining as normal.  This causes storage providers to miss blocks.  Details [here](https://github.com/filecoin-project/lotus/issues/7833).

There is a tradeoff here between security and fairness.  In particular SPs with worse network connectivity will be more likely to miss the full tipset before mining a block leading to the above situation.  It's not obvious to me that the current parent grinding consensus fault type is the right tradeoff.

So far @llifezou has reported this situation.  It would be good to learn if others have this problem.

Note that parent grinding faults only apply in the case where a miner wins two blocks in a row, so in normal operation this is a relatively rare event.  Even if we agree that these faults do more harm than good it's possibly not worth the effort to do anything about it.

**Comment by steven004** [comment]
Agreed. We do not need to do anything about it, not only because this is relatively rare, but we do not really know if a storage provider is out of sync, or intensionally just choose some blocks as parents to mine.  That is, if we loose the check, a storage provider could attack the network by forking the chain via mining on selected blocks.
